### PR TITLE
rename rdp to tcp

### PR DESCRIPTION
**Pull request by mustafakirimli**

### CODE REVIEW

File: cmd/cloudflared/tunnel/cmd.go [hunk 1]
```diff
@@ -546,7 +546,7 @@ func hostnameFromURI(uri string) string {
 	switch u.Scheme {
 	case "ssh":
 		return addPortIfMissing(u, 22)
-	case "rdp":
+	case "tcp":
 		return addPortIfMissing(u, 3389)
```
Comment (sodabrew) [inline cmd/cloudflared/tunnel/cmd.go]: This port, 3389, is for the RDP protocol: https://en.wikipedia.org/wiki/Remote_Desktop_Protocol

Comment (sssilver) [reply cmd/cloudflared/tunnel/cmd.go]: Thanks for the PR, @mustafakirimli! Before we review the code (thanks @sodabrew!), do you mind creating an issue in the Github project outlining what this change is accomplishing and why?

**Comment by sodabrew** [comment]
Aha, I see what you're getting at now. It doesn't even make sense that `rdp` is implemented as an alias of `ssh` but with a different default port. As you add in this PR, `tcp` must be accompanied by a port number. It's not an alias of `ssh` because none of these other subcommands made any sense with a general `tcp` service. Suggest to split `ssh` and `tcp` apart.

For backwards compatibility sake, `rdp` should have to be equal to `tcp` + `port 3389`

**Comment by mustafakirimli** [comment]
> Thanks for the PR, @mustafakirimli! Before we review the code (thanks @sodabrew!), do you mind creating an issue in the Github project outlining what this change is accomplishing and why?

Kindly find the issue below, 
@sssilver 
https://github.com/cloudflare/cloudflared/issues/130

**Comment by judge2020** [comment]
Keeping RDP as its own protocol might help in the future if/when Argo Tunnel supports proxying UDP, at which time an `rdp://` protocol would mean having a tunnel set up for both UDP and TCP port 3389.

**Comment by daltoniam** [comment]
Thanks for the PR! This is now officially supported: https://developers.cloudflare.com/access/other-protocols/tcp-guide/

**Pull request closed without merging**